packages/vue-components/src/Overlay.vue

```diff
@@ -0,0 +1,185 @@
+      @click="toggleNavMenu"
+    ></span>
+    <div ref="navMenuContainer" :class="[navMenuIconClass, { 'nav-menu-open': show }]">
+      <retriever
+        v-if="hasIdentifier"
+        :src="src"
+        :fragment="fragment"
+        @src-loaded="navMenuLoaded"
+      />
+      <div v-else ref="navigationMenu"></div>
+    </div>
+  </div>
+</template>
+
+<script>
+import retriever from './Retriever.vue';
+import $ from './utils/NodeList';
+import { publish, subscribe } from './utils/pubsub';
+
+export default {
+  components: {
+    retriever,
+  },
+  props: {
+    type: {
+      type: String,
+      default: null,
+    },
+    src: {
+      type: String,
+      default: null,
+    },
+    fragment: {
+      type: String,
+      default: null,
+    },
+    hasIdentifier: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  inject: ['toggleLowerNavbar'],
+  data() {
+    return {
+      show: false,
+    };
+  },
+  computed: {
+    buttonId() {
+      switch (this.type) {
+      case 'siteNav':
+        return 'toggle-site-nav-button';
+      case 'pageNav':
+        return 'toggle-page-nav-button';
+      default:
+        return '';
+      }
+    },
+    navMenuIconClass() {
+      switch (this.type) {
+      case 'siteNav':
+        return 'site-nav-menu';
+      case 'pageNav':
+        return 'page-nav-menu';
+      default:
+        return '';
+      }
+    },
+  },
+  methods: {
+    toggleNavMenu() {
+      this.show = !this.show;
+      switch (this.type) {
+      case 'siteNav':
+        publish('closePageNav');
+        break;
+      case 'pageNav':
+        publish('closeSiteNav');
+        break;
+      default:
+        break;
+      }
+    },
+    navMenuLoaded() {
+      this.toggleLowerNavbar();
+      $(this.$refs.navMenuContainer).find('a').on('click', () => {
+        this.toggleNavMenu();
+      });
+    },
+  },
+  mounted() {
+    const navMenu = this.$refs.navigationMenu;
+    const buildNav = (navMenuItems) => {
+      for (let i = 0; i < navMenuItems.childNodes.length; i += 1) {
+        navMenu.appendChild(navMenuItems.childNodes[i].cloneNode(true));
+      }
+      this.navMenuLoaded();
+    };
+    const buildPageNav = () => {
+      const wrapper = document.createElement('div');
+      const pageNavTitle = document.getElementsByClassName('page-nav-title')[0];
+      const pageNavLinks = document.getElementById('mb-page-nav');
+      if (pageNavTitle) {
+        wrapper.appendChild(pageNavTitle.cloneNode(true));
+      }
+      if (pageNavLinks) {
+        wrapper.appendChild(pageNavLinks.cloneNode(true));
+      }
+      buildNav(wrapper);
+    };
+    const buildSiteNav = () => {
+      const siteNav = document.getElementsByClassName('site-nav-root')[0];
+      buildNav(siteNav);
+    };
+
+    switch (this.type) {
+    case 'siteNav':
+      if (!this.hasIdentifier) { buildSiteNav(); }
+      subscribe('closeSiteNav', () => { this.show = false; });
```

Comment:
would a generic event work? (e.g. `closeOtherOverlays`)

so no need to tie `overlay` to site / page nav specifically

Reply:
Thanks! Updated :)